packages/cli/src/commands/build.ts

```diff
@@ -585,15 +623,20 @@ async function doBuild(
   const mergedOverrides: Record<string, PathOverride> =
     overrides.length > 0 ? Object.assign({}, ...overrides) : undefined;
 
+  if (!framework) {
+    // if no `buildResult` identified a framework version, try to determine it here
+    framework = await determineFrameworkVersion(detectedFramework, cwd, output);
+  }
```

Comment:
Builders can return a framework version, but if they don't, we run framework detection to find the framework record, then leverage the new `versionDependencies` key to look up installed framework version.